### DISCUSSION TITLE
FoxESS modbus RS485

### DISCUSSION THREAD

**Discussion by Oliris123** [StealthChesnut/HA-FoxESS-Modbus#130] [Q&A]
Hi everyone!

First of all, happy new year for 2024!!!

I started working with Home Assistant on the Orange Pi 5 Plus two weeks ago and now want to expand my installation. I currently have solar panels with a FoxESS-H1-4.6-E_cube inverter and a FoxESS Energy Cube CS2900 battery.

Unfortunately, it's no longer possible to connect directly for data, so I explored the option of using an RS485 Modbus and combining it with the GitHub code. I have three questions before I proceed:

Is it possible to achieve this with the current hardware, or is there a better hardware option?
[Link to hardware on AliExpress](https://www.aliexpress.com/item/32916128353.html?spm=a2g0o.cart.0.0.141038daxutaLO&mp=1)

After setting this up, will it always be possible to view the data on the FoxESS cloud, meaning on the 2 systems: Home assistant and FoxESS cloud?

Is it also possible to display battery charging, capacity, etc.?

Thank you in advance for your support on this!

**Comment by Daibutt** [comment]
Try this;
https://github.com/nathanmarlor/foxess_modbus/wiki

Has lots of good info in it.

**Reply by Oliris123** [reply]
Thanks Daibutt!
I will check later and give feedback!

**Reply by Oliris123** [reply]
Hi Daibutt,

I checked today for this options and saw that the port for the RS485 is already used.
Probably connected to the smart meter. (needed in Belgium)
I think that I have no options to have my system to Home assistant?!
Do I have any other options?

Thanks in advance for your support!

Regards,
Oliris

**Reply by Oliris123** [reply]
![20240105_180459](https://github.com/StealthChesnut/HA-FoxESS-Modbus/assets/154283062/0ae6deb4-08f6-4f4d-ab2e-cf3ae02d4868)
![20240105_181452](https://github.com/StealthChesnut/HA-FoxESS-Modbus/assets/154283062/693cd972-d6d7-4ccc-ab99-c40c4c5c36c4)

Sorry, I forgot to send the picture!

**Reply by Malcs91TF** [reply]
If you are lucky the meter is connected to Pins 1 & 2 leaving you pins 3 and 4 for home assistant. 
![RS485](https://github.com/StealthChesnut/HA-FoxESS-Modbus/assets/16632008/e621caaa-297a-4b45-bf49-2420a6c23d70)

**Reply by Oliris123** [reply]
HI Malcs91TF,

Thanks for your reply!, I think that I can simply unplug this jig without power off the inverter?
Very appreciate Github!

Thanks,
Oliris123

**Reply by Oliris123** [reply]
Hi Malcs91TF,
Hi Daibutt,

Just controlled as from you purposed @Malcs91TF (thanks) and indeed pin 3 and 4 are free.
Can I consider in this way that I can integrate my installation to Home Assistant?
What is the best Hardware to use via wifi?:
[Elfin EW11]
[USR-TCP232-304]
[USR-TCP232-410s]
[USR-W610]
[Waveshare RS485 to ETH (B)]
Other...

After I connect the modbus, can I also follow the data's on the FOXess cloud? (No impact?)

Many thanks for your time and support!
Oliris123

![20240106_121759](https://github.com/StealthChesnut/HA-FoxESS-Modbus/assets/154283062/8fed7517-661e-4025-a7eb-42a074f5fc53)

**Reply by Daibutt** [reply]
There are some pros/cons for the various adapters listed in the wiki I referenced above. 
I believe the Elfin or Waveshare are the preferred ones these days. 
I use the W610 as we were just trying to get something to talk to the H1 inverter about 18months ago. 
It worked so I stuck with it, but it does have some limitations apparently as mentioned in the wiki. I do not have a load of sensors all polling at high frequency, some are set at 120 secs+ (temps, daily/annual totals etc.) Only things like PV power, load, Grid power are down at 5-7 seconds.

Yep, Foxcloud (i.e. www.foxcloud.com) and the mobile app will continue to work as they are totally separate.
Although if you currently have the Foxcloud HA integration and then install Nathan's integration, there may be some duplicate sesnor names so the new ones will get appended _2. If I was swapping from HA Foxcloud to Nathan's HA Modbus I would make a HA backup, renmove the Foxcloud integration, reboot HA, then install HA Modbus.

P.S. I would personally power down PV, baterries, BMS, inverter before removing/modifying the CT/RS485 connector.
https://www.youtube.com/watch?v=xBTtKaX-k14

**Reply by Oliris123** [reply]
Hi Daibutt,

I just ordered a ELFIN EW11A.
I get you updated when delivered and that I start with setup.

Best Regards,
Oliris123

**Comment by Malcs91TF** [comment]
Hi Oliris,
Personally I use the USR-W610 with my H1-3.7 inverter and it seems to be working well, I bought and installed the W610 about 12 months ago before seeing all the advice to use something else. 
Yes, as said by Daibutt, Foxcloud is still working and allows me to view data when I'm away from home as my remote connection to Home Assistant is unreliable.
I would have said that you must isolate the AC feed to the inverter before disconnecting this plug as the CT which controls the inverter uses the same plug but strangely I cannot see your CT wires in the photo.  However fully agree: Power down all the system before any disconnections of cabling, and if you disconnecting the CT cable you must open the CT clip before unplugging it's cable (remember to fit it the right way around after).  I do not know but perhaps the meter is acting in place of your CT. 
Regards Malcolm

**Comment by Oliris123** [comment]
Hi @Daibutt, 
Hi @Malcs91TF, 

Yesterday my Elfin EW11 received and installed today.
Everything work perfect due to your support, Many thanks!!!
Total price for this, 14€  for Elfin EW11 + USB power wiring.
This installation is made on my inverter FoxESS-H1-4.6-E cube combined with FoxESS energy cube CS2900.
To realize this I only followed all the steps on this link: [https://github.com/nathanmarlor/foxess_modbus/wiki](url)
This,is the result (not yet customized) fill free if you have some good links or ideas :

![image](https://github.com/StealthChesnut/HA-FoxESS-Modbus/assets/154283062/8485783e-4539-40a3-863f-6b5eb27ac47d)

![image](https://github.com/StealthChesnut/HA-FoxESS-Modbus/assets/154283062/0de1f979-2fa7-418d-8a72-ffb27a1c0858)

![image](https://github.com/StealthChesnut/HA-FoxESS-Modbus/assets/154283062/b9b13eb4-76ba-4521-8055-6e6642112c77)

![image](https://github.com/StealthChesnut/HA-FoxESS-Modbus/assets/154283062/4de9c944-eb6b-4cdd-8d94-3807fff7039c)


Regards,
Oliris123
